### PR TITLE
[F] Authentication improvements

### PR DESCRIPTION
**Pull request by dananjohnson**

### CODE REVIEW

**Comment by dananjohnson** [comment]
@jen-castiron would you mind testing out all the user journeys through authentication and seeing what you think? Thanks!

File: hooks/useAuthentication.js [hunk 1]
```diff
@@ -130,97 +211,93 @@ export default function useAuthentication() {
...
+    const data = await authenticateFacebook({ code, pendingGroup });
 
     const returnRole =
-      role === "teacher" ? "googleSignInTeachers" : "googleSignInStudents";
-
-    if (data?.[returnRole]) {
-      setStateFromResponse(data[returnRole]);
-    }
+      pendingGroup === "teachers"
+        ? "facebookSignInTeachers"
```
Comment (jen-castiron) [inline hooks/useAuthentication.js:280]: When I try to register as a teacher, I'm getting this error: 
```
"Cannot query field "facebookSignInTeachers" on type "Mutation". Did you mean "facebookSignInEditors", "facebookSignInEducators", or "facebookSignInStudents"?"
```
If the schema has changed to "Editors" then the google & facebook mutations will need to update too.

I also had trouble signing in as a student with facebook/google:
```
"No schema has been set for this user group"
```
Screen Shot 2022-03-16 at 1 31 51 PM

**Review by jen-castiron — COMMENTED** [review]
@dananjohnson This feels a lot smoother, but there's a few little bugs. (If you're discussing them with Kyle right now, ignore my comments 😅 )

**Comment by dananjohnson** [comment]
Thanks for reviewing, @jen-castiron! Yeah, I need to update everything from Teachers to Educators now that Kyle has pushed that change to the api repo. I'll do that and then merge.

That last error is from a plugin config issue in the backend. Seems like the schema was lost on a recent commit.

**Pull request merged**